### PR TITLE
LPD-253 Healthchecks

### PR DESCRIPTION
**Pull request by olafk**
Now without any SF complaint: 
* 1 commit for API
* 1 commit for trivial initial Healthchecks (no test)
* 1 commit for a sample UI, not meant for inclusion in master

### CODE REVIEW

**Comment by olafk** [comment]
ci:test:sf

**Comment by olafk** [comment]
ci:test:sf

File: modules/apps/healthcheck/healthcheck-api/src/main/java/com/liferay/healthcheck/HealthcheckItem.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+				if (result == null) {
+					result = key;
+				}
+			}
+		}
+
+		return result;
+	}
+
+	private final Healthcheck _healthcheck;
```
Comment (shuyangzhou) [inline modules/apps/healthcheck/healthcheck-api/src/main/java/com/liferay/healthcheck/HealthcheckItem.java:135]: Healthcheck is a service, HealthcheckItem is a data object. A data object should not hold a reference to service, that would be considered as a service leaking. I guess you want to group all HealthcheckItem from one Healthcheck together, but that should be done explicitly at the web layer when consuming the Healthcheck service. Please remove this field.

File: modules/apps/healthcheck/healthcheck-api/src/main/java/com/liferay/healthcheck/HealthcheckItem.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	/**
+	 * signals if the healthcheck result is healthy or not
+	 *
+	 * @return true if healthy
+	 */
+	public boolean isResolved() {
+		return _resolved;
+	}
+
+	private String _lookup(Locale locale, String key, Object... parameters) {
```
Comment (shuyangzhou) [inline modules/apps/healthcheck/healthcheck-api/src/main/java/com/liferay/healthcheck/HealthcheckItem.java:109]: Localization logic should all be moved to web layer. It is not API layer's duty to do this.

**Review by shuyangzhou — COMMENTED** [review]
Also please clean up all build.gradle's dependencies list, there are a lot unused ones. Make sure all dependencies you state are truly used.

**Comment by shuyangzhou** [comment]
@olafk please see comment above, I only checked the api bundle. Please fixes those 3 issues, then I will resume the review. Thanks!

**Comment by olafk** [comment]
Sorry about that: The whole SF workload that I wasn't used to let me lose track over the localization still being in the wrong place. Changed, and I like the API better now.
The massive dependencies came from the other (uncommitted) healthchecks in the projects. I hope to have brought them down to a bare minimum now.

Can I put ci:test:sf in this comment as well?

**Comment by shuyangzhou** [comment]
@olafk please squash the fix commits to the original ones. the general rule is, if all commits are from you, then you should not have a following commit fixing stuff from previous commit. As that would be considered unnecessary history. But once someone else adds commits after your change, you should avoid amend your previous commits, just append new ones to make changes.

And yes, you can run a ci task in the same comment message. but the ci task must be at the very beginning of the message in its own line. otherwise CI will just see it as a regular text content in your message.

**Comment by olafk** [comment]
Before having to deal with force-push for amended commits to a branch that's already pullrequested, I've rewritten those changes in a new branch. I'll close this PR and try again with the new branch.

**Pull request closed without merging**